### PR TITLE
Fix #132 - Add encrypt-data-using-hc-128

### LINKED ISSUE #132: encrypt-data-using-hc-128
## Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Checked that your rule idea isn't already filed: [search](https://github.com/fireeye/capa-rules/issues?q=is%3Aissue+is%3Aopen+)

## Summary

encrypt-data-using-hc-128

## Examples

e69a8eb94f65480980deaf1ff5a431a6 from https://download.bitdefender.com/resources/files/News/CaseStudies/study/318/Bitdefender-TRR-Whitepaper-Maze-creat4351-en-EN-GenericUse.pdf

## Features

There don't appear to be any unique constants used by this algorithm, but the following pieces seem like they will be unique enough when combined to work.

From https://github.com/rost1993/hc128/blob/master/hc128.c
```
#define ROTR32(v, n)	((v >> n) | (v << (32 - n)))
...
#define F1(x)		(ROTR32(x,  7) ^ ROTR32(x, 18) ^ (x >>  3))
#define F2(x)		(ROTR32(x, 17) ^ ROTR32(x, 19) ^ (x >> 10))
...
static void
hc128_initialization_process(struct hc128_context *ctx)
...
	for(i = 16; i < (256 + 16); i++)
		ctx->w[i] = F2(ctx->w[i-2]) + ctx->w[i-7] + F1(ctx->w[i-15]) + ctx->w[i-16] + i;
```

### Namespace

Nursery for now, per https://github.com/fireeye/capa-rules/issues/131#issuecomment-695590953

### PR DESCRIPTION
**Pull request by recvfrom**
Fix #132 - Add encrypt-data-using-hc-128

### CODE REVIEW

File: nursery/encrypt-data-using-hc-128.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+rule:
+  meta:
+    name: encrypt-data-using-hc-128
+    namespace: data-manipulation/encryption/hc-128
+    author: '@recvfrom'
+    description: Looks for instruction mnemonics associated with initialization of the HC-128 stream cipher
+    scope: basic block
+    examples:
```
Comment (mr-tz) [inline nursery/encrypt-data-using-hc-128.yml:8]: can you please add this ATT&CK info here?
```yaml
att&ck:
      - Defense Evasion::Obfuscated Files or Information [T1027]
```

**Review by mr-tz — COMMENTED** [review]
Very good, thanks a lot for the rule and extensive documentation! If you have a moment, can you please add the ATT&CK info before merging.

I'm not too worried about FPs as this is a tight rule given the BB scope and features.

File: nursery/encrypt-data-using-hc-128.yml (hunk 1 above)
Comment (recvfrom) [reply nursery/encrypt-data-using-hc-128.yml:8]: Will do! I too feel pretty good about the rule after adding in the mnemonic counts.  Should the rule be moved out of the nursery, then?

Comment (mr-tz) [reply nursery/encrypt-data-using-hc-128.yml:8]: Yes, please!

**Review by mr-tz — APPROVED** [review]
awesome, thank you!

**Pull request merged**